File: 1/1-7.ts
function sqrtIter(guess: number, x: number): number {
    if (Math.abs(guess * guess - x) < 0.001) {
        return guess;
    } else {
        return sqrtIter((guess + x / guess) / 2, x);
    }
}

function squareRoot(x: number): number {
    return sqrtIter(1.0, x);
}

console.log(squareRoot(2.0));

File: 1/1-7.py
def sqrt_iter(guess, x):
    if abs(guess * guess - x) < 0.001;
        return guess
    else:
        return sqrt_iter((guess + x / guess) / 2, x)

def square_root(x):
    return sqrt_iter(1.0, x)

print(square_root(2.0))